enhancements/oc/mirroring-manifests.md

```diff
@@ -0,0 +1,260 @@
+
+If `oc` encountered an error applying a manifest it should warn the user, and then either:
+
+* Immediately exit or
+* Continue to apply remaining manifest to the cluster.
+
+In either case, it would exit non-zero.
+
+### Pushing to disk
+
+As of 4.3 (with [this change][relative-to-dir-value] to support relative `--to-dir` values):
+
+```console
+$ oc adm release mirror --to-dir=mirror quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+...
+Success
+Update image:  openshift/release:4.3.0
+
+To upload local images to a registry, run:
+
+    oc image mirror --from-dir=mirror file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+```
+
+creates an output filesystem like:
+
+```console
+$ tree mirror
+mirror
+└── v2
+    └── openshift
+        └── release
+            ├── blobs
+            │   ├── sha256:005337dc1f6870934cf5efaf443786149e6a71041a91566ce1e16e78bce511a9
+            ...
+            │   └── sha256:ffc132fefac522ec38b1b22ecd67a5173565efb3e8929e47ef7e0b4ee7920adf
+            └── manifests
+                ├── 4.3.0 -> sha256:3a516480dfd68e0f87f702b4d7bdd6f6a0acfdac5cd2e9767b838ceede34d70d
+                ...
+                ├── 4.3.0-thanos -> sha256:42ab3f59d3e769e88af89c30304b9cc77a76b66960747e2e1a6f8db097420858
+                ├── sha256:00ec58112784d340179e045bc70399ac1bd509ae43c4411e934ced23d016b2a1
+                ...
+                └── sha256:ffa87c992c5a2c51290e6d67cb609f29893dd147c31cff639ff6785ae7a1cfe2
+
+5 directories, 514 files
+```
+
+This filesystem can then be applied to the local registry with the given `oc image mirror` command.
+
+With this enhancement, the output format would gain an additional directory `manifests` under the base directory, a sibling to the current `v2`, containing manifest files that can be applied to the local cluster with:
+
+```console
+$ oc apply -Rf mirror/manifests
+```
+
+The `manifests` directory may or may not contain additional subdirectories, so the recursive `-R` should be given for future-proofing.
+
+### Pushing to disk while pushing containers to the mirror registry
+
+Users mirroring a release before creating a cluster may want to [push both container images and manifests to disk](#pushing-to-disk).
+But they might also want to push container images directly to a mirror repository (as in [the *applying directly to the target cluster* case](#applying-directly-to-the-target-cluster)) while still pushing manifests to disk (because there is not yet a cluster into which those manifests can be pushed).
+While you could address this use-case with [the *pushing to disk* flow](#pushing-to-disk) followed by an immediate `oc image mirror ...`, it is more convenient to have `oc adm release mirror ...` push to the registry directly.
+Because setting `--to-dir` adjusts the image source, this enhancement extends the command with a new `--manifests-to-dir` that can be used to set a manifest output directory without setting a container image output directory.
```

Comment:
`--to-manifests-dir` is consistent with our naming patterns (`--to-dir`, `--to-release`)

Reply:
> `--to-manifests-dir` ...

I went with `--kubernetes-manifests-to-dir`, making the content (Kubernetes manifests) and to-directory-ness explicit, but leaving the name of the target directory implicit.  The target directory *could* also contain other things, so it's not necessarily a directory only for Kubernetes manifests.  Thoughts?

Reply:
I'm usually uncomfortable with the word kubernetes anywhere.  We use "manifest" consistently within `oc adm release` to describe "Kubernetes YAML files that are part of a release".  I guess a distinction we could have here is whether these are even more specialized (those are "release manifests", these are "configuration manifests").  So we could potentially use `--to-config-manifest-dir` (config is our general term for yaml applied to a cluster).  But in that case we can drop `manifest` too and just say `--to-config-dir: Directory to output configuration manifests intended for the cluster that will use this mirror.`

Reply:
> I'm usually uncomfortable with the word kubernetes anywhere.

Pivoted to "configuration manifests", a `config` directory, and `--apply-config` and `--config-to-dir` options with b2f6e8a -> 7dcc2b9.